cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -273,6 +273,31 @@ class alignas(16) column_device_view : public detail::column_device_view_base {
     return data<T>()[element_index];
   }
 
+  /**---------------------------------------------------------------------------*
+   * @brief Iterator for navigating this column
+   *---------------------------------------------------------------------------**/
+  template <typename T>
+  struct value_accessor;
+  using count_it = thrust::counting_iterator<size_type>;
+  template <typename T>
+  using const_iterator = thrust::transform_iterator<value_accessor<T>, count_it>;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Return new iterator pointing to the beginning of this column
+   *---------------------------------------------------------------------------**/
+  template <typename T>
+  const_iterator<T> begin() const {
+    return const_iterator<T>{count_it{0}, value_accessor<T>{*this}};
```

Comment:
This almost certainly can't work. `value_accessor` is an incomplete type (it's only forward declared), yet it's constructor is being invoked here.

The definition of `value_accessor` needs to be accessible in this file, either by moving it to this file or including `iterator.cuh`.

Reply:
moved.